src/app/component/MissionDetailsCard.jsx

```diff
@@ -0,0 +1,384 @@
+
+const StyledButton = styled(Button)`
+  flex-grow: 1;
+`;
+
+const useStyles = makeStyles((theme) => ({
+  content: {
+    padding: theme.spacing(2),
+  },
+  goBackIcon: {
+    fontSize: 32,
+    fill: color.deepPurple,
+    "&:hover": {
+      cursor: "pointer",
+    },
+  },
+  cardHeader: {
+    paddingBottom: theme.spacing(1),
+  },
+  subheader: {
+    marginTop: theme.spacing(0.3),
+  },
+  contentTypography: {
+    paddingLeft: theme.spacing(0.5),
+  },
+  avatar: {
+    height: "25px",
+    width: "25px",
+    marginBottom: theme.spacing(0.5),
+  },
+  image: {
+    height: 0,
+    paddingTop: "56.25%", // 16:9
+  },
+  missionTypeText: {
+    paddingTop: theme.spacing(0.5),
+  },
+  deliveryDetailsHeader: {
+    paddingTop: theme.spacing(1),
+    fontWeight: 600,
+  },
+  deliveryDetails: {
+    marginTop: theme.spacing(0.5),
+  },
+  unassignButton: {
+    color: color.darkPink,
+    textDecoration: "underline",
+  },
+}));
+
+const titleCase = (str) => ("" + str).charAt(0).toUpperCase() + ("" + str).substr(1);
+
+const MissionDetailsStatus = ({ status, volunteerName }) => {
+  return status === null ? null : status === MissionStatus["unassigned"] ? (
+    titleCase(status)
+  ) : status === MissionStatus["tentative"] || status === MissionStatus["assigned"] ? (
+    volunteerName
+  ) : (
+    <>
+      {volunteerName} - <b>{titleCase(status)}</b>
+    </>
+  );
+};
+
+const MissionDetailsIconList = ({ contentItems, classes, outerClass }) => (
+  <Grid container className={outerClass} alignItems="flex-start">
+    {contentItems.map((contentItem, index) => {
+      const Icon = contentItem.icon;
+      const avatarImage = contentItem.avatar?.image;
+      const content = contentItem.content.map((content, index) => {
+        return (
+          <Body2
+            key={`content-item-txt-${index + 1}`}
+            color="textPrimary"
+            style={content.style}
+            className={classes.contentTypography}
+          >
+            {content.text}
+          </Body2>
+        );
+      });
+
+      return (
+        <React.Fragment key={`content-item-${index + 1}`}>
+          <Grid item xs={1}>
+            {Icon !== undefined ? <Icon color="primary" /> : null}
+            {avatarImage !== undefined ? (
+              <Avatar className={classes.avatar} alt="Volunteer" src={avatarImage} />
+            ) : null}
+          </Grid>
+          <Grid item xs={11}>
+            {content}
+          </Grid>
+        </React.Fragment>
+      );
+    })}
+  </Grid>
+);
+
+const MissionDetailsContent = ({ description, classes }) => (
+  <Box>
+    <H5 align="left" color="textSecondary">
+      Mission Type
+    </H5>
+    <Body2 align="left" className={classes.missionTypeText} color="textPrimary">
+      {description}
+    </Body2>
+  </Box>
+);
+
+const MissionDetailsButton = ({
+  type,
+  volunteerForMission,
+  startMission,
+  markMissionAsDelivered,
+}) => {
+  switch (type) {
+    case MissionStatus["unassigned"]:
```

Comment:
This is more readable:

```suggestion
    case MissionStatus.unassigned:
```